pkgs/by-name/yo/yofi/package.nix

```
@@ -0,0 +1,45 @@
+{ lib
+, fetchFromGitHub
+, rustPlatform
+, autoPatchelfHook
+, fontconfig
+, libxkbcommon
+, pkg-config
+, ...
+}:
+let
+  buildDeps = [ fontconfig libxkbcommon pkg-config ];
+in
+rustPlatform.buildRustPackage rec {
+  pname = "yofi";
+  version = "0.2.2";
+
+  src = fetchFromGitHub {
+    owner = "l4l";
+    repo = "yofi";
+    rev = version;
+    hash = "sha256-cepAZyA4RBgqeF20g6YOlZTM0aRqErw17yuQ3U24UEg=";
+  };
+
+  cargoHash = "sha256-iSy/y1iwhR8x3wDIfazMeROSrJ8uRyA10hoNo6y2OQc=";
+  buildInputs = buildDeps;
+  nativeBuildInputs = buildDeps;
+
+  postFixup = ''
+    patchelf $out/bin/yofi --add-rpath ${lib.makeLibraryPath buildDeps}
+  '';
```

There should usually be no overlap between `buildInputs`, `nativeBuildInputs`, and  manually adding to `RPATH`/`RUNPATH`, except for some really exceptional cases. Also `autoPatchelfHook` wasn't getting used at all, so added that in.

```suggestion
  nativeBuildInputs = [
    autoPatchelfHook
    pkg-config
  ];

  buildInputs = [
    libgcc
    libxkbcommon
  ];

  appendRunpaths = [
    (lib.makeLibraryPath [ fontconfig ])
  ];
```